crates/bevy_pbr/src/render/mesh.rs

```diff
@@ -1231,95 +1231,54 @@ pub fn prepare_mesh_view_bind_groups(
-                    ),
-                },
-                BindGroupEntry {
-                    binding: 3,
-                    resource: BindingResource::Sampler(&shadow_samplers.point_light_sampler),
-                },
-                BindGroupEntry {
-                    binding: 4,
-                    resource: BindingResource::TextureView(
-                        &view_shadow_bindings.directional_light_depth_texture_view,
-                    ),
-                },
-                BindGroupEntry {
-                    binding: 5,
-                    resource: BindingResource::Sampler(&shadow_samplers.directional_light_sampler),
-                },
-                BindGroupEntry {
-                    binding: 6,
-                    resource: point_light_binding.clone(),
-                },
-                BindGroupEntry {
-                    binding: 7,
-                    resource: view_cluster_bindings.light_index_lists_binding().unwrap(),
-                },
-                BindGroupEntry {
-                    binding: 8,
-                    resource: view_cluster_bindings.offsets_and_counts_binding().unwrap(),
-                },
-                BindGroupEntry {
-                    binding: 9,
-                    resource: globals.clone(),
-                },
-                BindGroupEntry {
-                    binding: 10,
-                    resource: fog_binding.clone(),
-                },
-                BindGroupEntry {
-                    binding: 11,
-                    resource: BindingResource::TextureView(
-                        ssao_textures
-                            .map(|t| &t.screen_space_ambient_occlusion_texture.default_view)
-                            .unwrap_or(&fallback_ssao),
-                    ),
-                },
-            ];
+            let mut entries = DynamicBindGroupEntries::sequential((
+                // 0
+                view_binding.clone(),
+                light_binding.clone(),
+                &view_shadow_bindings.point_light_depth_texture_view,
+                &shadow_samplers.point_light_sampler,
+                &view_shadow_bindings.directional_light_depth_texture_view,
+                // 5
+                &shadow_samplers.directional_light_sampler,
+                point_light_binding.clone(),
+                view_cluster_bindings.light_index_lists_binding().unwrap(),
+                view_cluster_bindings.offsets_and_counts_binding().unwrap(),
+                globals.clone(),
+                // 10
+                fog_binding.clone(),
+                ssao_textures
+                    .map(|t| &t.screen_space_ambient_occlusion_texture.default_view)
+                    .unwrap_or(&fallback_ssao),
+            ));
 
-            let env_map = environment_map::get_bindings(
+            // 12 - 14
+            entries = entries.extend_sequential(environment_map::get_bindings(
```

Comment:
Not sure if I'm a fan of this or not. It feels like existence of the comments implies that the number should be just written in. It's really critical that these binding numbers are obvious so that 3rd party stuff knows where to bind. I especially am not sure about the `extend_sequential`.

Reply:
I’ll change this to use the `with_indices` variants. I’ll leave the sequential api available though in case anyone wants it. 

hopefully this will change relatively soon to use automatic binding slot allocation anyway.